### DISCUSSION TITLE
Terminal hyperlinks in help output break easily because the 80 columns print is not aware of escape sequences

### DISCUSSION THREAD

**Discussion by i30817** [fastapi/typer#613] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def link(uri, label=None, parameters=''):
    '''
    Found in https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda , windows console and many unix consoles trick to embeed hyperlinks/uri with text
    '''
    if label is None:
        label = uri
    # OSC 8 ; params ; URI ST <name> OSC 8 ;; ST
    escape_mask = '\033]8;{};{}\033\\{}\033]8;;\033\\'
    return escape_mask.format(parameters, uri, label)

import typer
from typing import Optional, List
def test(address: Optional[str] = typer.Option('http://localhost.com', metavar='URL', help=f'URL of {link("https://localhost.com/bigurl-toobiggggggggggggggggg", "the biggest url you have ever seen ever")}.')):
  pass

if __name__ == "__main__":
    typer.run(test)

#run as 'python3 -m test --help'
```

### Description

I was wondering it it would be possible to make the print facility of typer when --help is invoked to ignore the length and not break on the middle, of escape sequences so that tricks like the above (placing a url link in the help output of a command for a option) will work.

### Operating System

Linux

### Typer Version

0.5.0

### Python Version

Python 3.10.6

**Comment by Danthewaann** [comment, marked as answer]
In rich mode (`rich_markup_mode="rich"`) you can now just use `[link=<url>]<display_text>[/link]` to create a hyperlink, like so:

```python
import typer


URL = "https://example.com/my-docs.md"
app = typer.Typer(
    name="my-command",
    help="My command help",
    epilog=f"More documentation: [link={URL}]{URL}[/link]",
    rich_markup_mode="rich"
)
```
